tools/duplicate_function_eliminator.py

```diff
@@ -0,0 +1,428 @@
+
```

Comment:
This file looks like a fork of `js_optimizer.py`? It has duplicated globals like `NUM_CHUNKS_PER_CORE` as well as classes like `Minifier`, functions like `split_funcs`, etc.? Could we import `js_optimizer.py` and reuse those parts, and only have unique and DFE-specific code here?